model/annotation.go

```diff
@@ -17,7 +17,23 @@ func (a *Annotation) MarshalJSON() ([]byte, error) {
 		Timestamp int64  `json:"timestamp"`
 		Value     string `json:"value"`
 	}{
-		Timestamp: a.Timestamp.UnixNano() / 1e3,
+		Timestamp: a.Timestamp.Round(time.Microsecond).UnixNano() / 1e3,
 		Value:     a.Value,
 	})
 }
+
+// UnmarshalJSON implements custom JSON decoding
+func (a *Annotation) UnmarshalJSON(b []byte) error {
+	type Alias Annotation
+	annotation := &struct {
+		TimeStamp int64 `json:"timestamp,omitempty"`
```

Comment:
does omitempty make sense? the field is mandatory...

Reply:
we can remove omitempty tag... automatic behavior on my part...